core/src/main/java/org/arquillian/smart/testing/SystemPropertyResolver.java

```diff
@@ -0,0 +1,33 @@
+package org.arquillian.smart.testing;
+
+public class SystemPropertyResolver {
+
+    final String previousCommit;
+    final String commit;
+
+    public SystemPropertyResolver() {
+        String prevDefaultValue = getPrevDefaultValue();
+        previousCommit = System.getProperty("git.previous.commit", prevDefaultValue);
+        commit = System.getProperty("git.commit", "HEAD");
+    }
+
+    public String getPreviousCommit() {
+        return previousCommit;
+    }
+
+    public String getCommit() {
+        return commit;
+    }
+
+    private String getPrevDefaultValue() {
+        String noOfCommits = System.getProperty("git.lastcommits");
```

Comment:
You can also set it as default "0" then `HEAD~0` will work the same way, except of being in the detached state.